Dev

desht opened the pull request:
Couple of fixes and added same abstraction layers as for 1.19

LatvianModder commented:
Why is there `+import javax.xml.soap.Text;`? That isn't used anywhere as far as I can tell

MinecraftXwinP commented:
opss, Sorry, that was misadded by the IDE.

LatvianModder commented:
Eh, still not a fan of implementation. I think I will write my own and just mention you in the commit if that's fine with you 😛  I wanted this to be implemented in every GuiButtonList base and I thought of a simplier way to do it

MinecraftXwinP commented:
OK, feel free to close the PR. I think the implementation is a little by hacky too.

desht committed:
fix: KeyMappingAccessor is a client mixin

desht committed:
build: version -> 2001.1.3

desht committed:
feat: added gamestages abstraction layer

Includes default fallback implementation using entity tags

Will replace direct stages integration code in other mods (quests, chunks...)

desht committed:
feat: added permission provider abstraction layer

Includes fallback (no-op) implementation.
Actual implementations to be provided by mods.

desht committed:
chore: a few changes to help with FTB Quests 1.20 porting

Make a couple of methods public, add an accessor for AtlasSpriteIcon id

desht committed:
fix: ensure serverconfig/ folder exists when copying default configs

On Fabric, it may not yet exist when creating a new world, since the Fabric
server-starting event (via Architectury) happens earlier than the equivalent
Forge event

https://github.com/FTBTeam/FTB-Mods-Issues/issues/867
